packages/apollo-server-core/src/caching.ts

```diff
@@ -0,0 +1,12 @@
+export interface KeyValueCache {
+  set(
+    key: string,
+    data: string,
+    options?: {
+      ttl?: number;
+      tags?: string[];
+    },
+  ): Promise<void>;
+  get(key: string): Promise<string>;
```

Comment:
What does this API do if the item is not found in the cache? Is that trivially distinguishable from "error talking to the cache" and "cache value is the empty string"?  `Promise<string | null>` would be one way to do this.